src/utils/liquidityPools.js

```diff
@@ -123,18 +123,33 @@ export const getAddLiquidityTransactions = async (
       : null;
 
     if (!approvedAmountBN || tokenAmountBN.gt(approvedAmountBN)) {
-      const approveTransactionData =
-        buildERC20ApproveTransactionData(ADDRESSES.router, tokenAmount, tokenAsset.decimals);
       addLiquidityTransactions = [
         {
           from: sender,
           to: tokenAsset.address,
-          data: approveTransactionData,
+          data: buildERC20ApproveTransactionData(ADDRESSES.router, tokenAmount, tokenAsset.decimals),
           amount: 0,
           symbol: ETH,
         },
         ...addLiquidityTransactions,
       ];
+
+      // ERC20 token contracts (incl. PLR itself) can throw error when trying to change non-zero approved allowance
+      // in order to prevent spedning both old and new allowance. As a way to mitigate it, we are first sending
+      // zeroing approve transaction.
+      // See: https://docs.openzeppelin.com/contracts/2.x/api/token/erc20#IERC20-approve-address-uint256-
+      if (approvedAmountBN?.gt(0)) {
```

Comment:
I wonder if we should actually apply it only for PLR token. Wdyt?

Reply:
As far as I understood the ERC20 docs you cited yesterday, it's good practice to do this even if the token's contract doesn't explicitly require it like PLR. So I'd leave it for all tokens, wdyt @poocart ?
```
Beware that changing an allowance with this method brings the risk that someone may use both the old and the new allowance by unfortunate transaction ordering. One possible solution to mitigate this race condition is to first reduce the spender's allowance to 0 and set the desired value afterwards
```